arrays/arrays.js

```diff
@@ -0,0 +1,116 @@
+function isArrayWithArray(a) {
+  return Array.isArray(a);
+}
+
+function isArrayWithInstanceOf(a) {
+  return a instanceof Array;
+}
+
+function range() {
```

Comment:
1. where are your params?
2. if I calls range(10, 20, 5, 5, 5) it should works as range(10, 20, 5)
3. range(10) [0..9]
range(1, 10) [1..9]
range(1, 10, 3) [1, 4, 7]
range(10, null, 3) [0, 3, 6, 9]
range(10, null) the same with range(10)
range(-2, -5) []
range(-5, -2) [-5, -4, -3]
range(-5, -2, 2) [-5, -3]
range(-5, -2, -2) []
range(-5, null, 2) []
range(-5, null, -2) [0, -2, -4]
range(-10, -20, -5) [-10, -15]
range(-20, -10, -5)
4. too many conditions and loops

FYI firstly you should find step and length of result array